org.civicrm.hremergency/CRM/Hremergency/Service/EmergencyContactService.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+class CRM_Hremergency_Service_EmergencyContactService {
```

Comment:
If we consider the pear underscore as a namespace separator this would be like `CRM\Hremergency\Service\EmergencyContact` which I don't like because then we would have multiple classes of the same name.

`CRM\Hremergency\Validator\EmergencyContact`
`CRM\Hremergency\Service\EmergencyContact`

I think the class name itself should be descriptive without the namespace, so I would prefer to keep the `Service` suffix. Again using the PSR analogy, with use statements the class would appear as `EmergencyContact`, which on it's own doesn't make much sense to me.

Reply:
Even though I don't link the duplication in the classname/namespace, I don't have any strong opinions on this. Also, you have some good points and it might indeed get confusing in situations. Let's keep the name as it is then.